### ISSUE TITLE
Traccar not connecting to server.

### ISSUE THREAD

**Issue by thalward** [traccar/traccar-client-android#341]

**Comment by thalward** [comment]
I am able to send an instant location update in other apps so it is getting a fix ok. The only difference is that it doesnt seem to want to open the port and communicate with the server



Sent from my Samsung Galaxy A3 2017 powered by Three
-------- Original message --------From: Anton Tananaev <notifications@github.com> Date: 08/08/2018  00:20  (GMT+00:00) To: traccar/traccar-client-android <traccar-client-android@noreply.github.com> Cc: thalward <tom@halward.uk>, Author <author@noreply.github.com> Subject: Re: [traccar/traccar-client-android] Traccar not connecting to server. Possible Client Device Port Issue (#341) 
Have you checked status screen in the app?
If GPS icon is flashing, it means that there is no GPS fix.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/traccar/traccar-client-android","title":"traccar/traccar-client-android","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/traccar/traccar-client-android"}},"updates":{"snippets":[{"icon":"PERSON","message":"@tananaev in #341: Have you checked status screen in the app?\r\n\r\nIf GPS icon is flashing, it means that there is no GPS fix."}],"action":{"name":"View Issue","url":"https://github.com/traccar/traccar-client-android/issues/341#issuecomment-411233211"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/traccar/traccar-client-android/issues/341#issuecomment-411233211",
"url": "https://github.com/traccar/traccar-client-android/issues/341#issuecomment-411233211",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
},
{
"@type": "MessageCard",
"@context": "http://schema.org/extensions",
"hideOriginalBody": "false",
"originator": "AF6C5A86-E920-430C-9C59-A73278B5EFEB",
"title": "Re: [traccar/traccar-client-android] Traccar not connecting to server. Possible Client Device Port Issue (#341)",
"sections": [
{
"text": "",
"activityTitle": "**Anton Tananaev**",
"activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
"activitySubtitle": "@tananaev",
"facts": [

]
}
],
"potentialAction": [
{
"name": "Add a comment",
"@type": "ActionCard",
"inputs": [
{
"isMultiLine": true,
"@type": "TextInput",
"id": "IssueComment",
"isRequired": false
}
],
"actions": [
{
"name": "Comment",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueComment\",\n\"repositoryFullName\": \"traccar/traccar-client-android\",\n\"issueId\": 341,\n\"IssueComment\": \"{{IssueComment.value}}\"\n}"
}
]
},
{
"name": "Close issue",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueClose\",\n\"repositoryFullName\": \"traccar/traccar-client-android\",\n\"issueId\": 341\n}"
},
{
"targets": [
{
"os": "default",
"uri": "https://github.com/traccar/traccar-client-android/issues/341#issuecomment-411233211"
}
],
"@type": "OpenUri",
"name": "View on GitHub"
},
{
"name": "Unsubscribe",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 365696887\n}"
}
],
"themeColor": "26292E"
}
]

**Comment by thalward** [comment]
Just an update, I have used developer mode to falsify my location and even with this enabled it fails to open the port to communicate with the server therefore can only assume the app will not open the port to communicate.

I was able to confirm the falsified location on other apps as a test to ensure a GPS fix was established. I then disabled the function and the GPS fix returned to my exact location.

**Comment by tananaev** [comment]
What do you mean by "open the port to communicate"? Do you mean bind/listen to the port? Only servers listen to ports.

**Comment by thalward** [comment]
I would assume so (apologies for my terminology).

It seems as though when Traccar runs correctly, it establishes a secure connection on port 443. It also opens up a second port to communicate and provide the information to the server (port 5055) and it is this port that is not opening to allow the device to communicate. 

I would assume this port (5055) opens regardless so a 'connected' event appears within the server log and that only a successful location update will create a hex event in the log file with the device identifier and location etc.
I would still expect it to log as connected in the server log even if no GPS fix can be obtained. The fact it is not even connecting and no port is opening points to the device. 

Unfortunately if I knew more about the android os I could investigate and diagnose further but it seems like its either an issue with the device not opening the port or the software not asking the port to be opened.

If you are able to advise any other steps you want me to take I would be more than happy to give them a try. It may be an android os issue with the ROM but I do not have the knowledge to go into that.

**Comment by tananaev** [comment]
I don't think you understand it correctly. Traccar Client only connects to the server when it successfully got a location from Android OS.

**Comment by thalward** [comment]
Ok, if this is the case, do you know how to test the android os to see why traccar seems to think a location hasnt been obtained? It obtains fine in other apps such as google maps.

**Comment by thalward** [comment]
All i get is service started, network offline, network online.

**Comment by tananaev** [comment]
OK, that proves that your device doesn't return location info. You need to check:

- Accuracy settings in the app
- Location providers in Android settings
- Location permissions for the app
- Network connection and GPS fix status

**Comment by thalward** [comment]
I have just run it on a different ROM on the same device and it works ok. I have reinstalled the correct ROM for the device and it fails to update again. From my limited knowledge the baseband firmware may be at fault but I could be having a complete wild 'stab in the dark' moment! I knew it could be one or the other but its still a case of pinpointing it.

My personal opinion would suggest that the app should be able to connect without providing GPS data so you know the server connection is up. Just my personal opinion though. 

I've gone back to the supplier and asked them to check the firmware so I'll close the thread if I get information back from them confirming so.

Worked on:
Baseband = MOLY.WR8.W1315.MD.WG.MP.V36.P1, 2016/11/29 19:57
Kernel Version = 3.4.67 sw05@honyao-Series #1 Wed Sep 13 11:31:02 CST 2017
Build = MTK6572 
Custom Build = V0.4

Not Working on (Inrico):
Baseband = MOLY.WR8.W1315.MD.WG.MP.V4, 2018/05/08 16:31
Kernel Version = 3.4.5 sw05@honyao-Series #1 Tue May 29 16:52:20 CST 2018
Build = ALPS.JB3.MP.V1.8
Custom Build = V0.1
